webrtc/RTCRtpParameters-encodings.html

```diff
@@ -0,0 +1,396 @@
+    validateSenderRtpParameters(param);
+
+    const { encodings } = param;
+    assert_equals(encodings.length, 1);
+
+    // {} is valid RTCRtpEncodingParameters because all fields are optional
+    encodings.push({});
+    assert_equals(param.encodings.length, 2);
+
+    return promise_rejects(t, 'InvalidModificationError',
+      sender.setParameters(param));
+
+  }, `sender.setParameters() with mismatch number of encodings should reject with InvalidModificationError`);
+
+  promise_test(t => {
+    const pc = new RTCPeerConnection();
+    const { sender } = pc.addTransceiver('audio');
+    const param = sender.getParameters();
+    validateSenderRtpParameters(param);
+
+    param.encodings = undefined;
+
+    return promise_rejects(t, 'InvalidModificationError',
+      sender.setParameters(param));
+
+  }, `sender.setParameters() with encodings unset should reject with InvalidModificationError`);
+
+  promise_test(t => {
+    const pc = new RTCPeerConnection();
+    const { sender } = pc.addTransceiver('audio');
+
+    const param = sender.getParameters();
+    validateSenderRtpParameters(param);
+    const encoding = getFirstEncoding(param);
+    const { ssrc } = encoding;
+
+    // ssrc may not be set since it is optional
+    if(ssrc === undefined) {
+      encoding.ssrc = 2;
+    } else {
+      // If it is set, increase the number by 1 to make it different from original
+      encoding.ssrc = ssrc + 1;
+    }
+
+    return promise_rejects(t, 'InvalidModificationError',
+      sender.setParameters(param));
+
+  }, `setParameters() with modified encoding.ssrc field should reject with InvalidModificationError`);
+
+  promise_test(t => {
+    const pc = new RTCPeerConnection();
+    const { sender } = pc.addTransceiver('audio');
+
+    const param = sender.getParameters();
+    validateSenderRtpParameters(param);
+    const encoding = getFirstEncoding(param);
+    const { rtx } = encoding;
+
+    if(rtx === undefined) {
+      encoding.rtx = { ssrc: 2 };
+    } else if(rtx.ssrc === undefined) {
+      rtx.ssrc = 2;
+    } else {
+      rtx.ssrc += 1;
+    }
+
+    return promise_rejects(t, 'InvalidModificationError',
+      sender.setParameters(param));
+
+  }, `setParameters() with modified encoding.rtx field should reject with InvalidModificationError`);
+
+  promise_test(t => {
+    const pc = new RTCPeerConnection();
+    const { sender } = pc.addTransceiver('audio');
+
+    const param = sender.getParameters();
+    validateSenderRtpParameters(param);
+    const encoding = getFirstEncoding(param);
+    const { fec } = encoding;
+
+    if(fec === undefined) {
+      encoding.fec = { ssrc: 2 };
+    } else if(fec.ssrc === undefined) {
+      fec.ssrc = 2;
+    } else {
+      fec.ssrc += 1;
+    }
+
+    return promise_rejects(t, 'InvalidModificationError',
+      sender.setParameters(param));
+
+  }, `setParameters() with modified encoding.rtx field should reject with InvalidModificationError`);
```

Comment:
Oops, think I patched that on a different computer and forgot to sync with local computer before force pushing. Have fixed that again.